common/types/hashable.go

```diff
@@ -0,0 +1,13 @@
+package types
+
+// A chain-agnostic generic interface to represent the following native types on various chains:
+// PublicKey, Address, Account, BlockHash, TxHash
+//
+//go:generate mockery --quiet --name Hashable --output ./mocks/ --case=underscore
+type Hashable[T any] interface {
+	MarshalText() (text []byte, err error)
+	UnmarshalText(text []byte) error
+	String() string
+	Equals(t T) bool
+	Empty() bool
```

Comment:
Do we have a story to follow up on `Empty() bool`? At the very least, we need to document the expectation for implementors.

Reply:
hmmmm I think empty won't be necessary anymore? there's a `nil` check for certain types but that shouldn't be needed once everything is just using copies rather than pointers for methods

and then this could become `Zero()`. I can add that to the story if that's the direction we want to go?

Reply:
Yeah so zero is well defined, but if we are using generic value types, then we could check against the literal zero value trivially, without needing an extra method.
Otherwise, I lost the thread, but I was thinking something up to the implementation to define, like `IsValid()`, would be more clear.

Reply:
ah good point. If i remember correctly, this method is only used to replace comparing against a zero address. So I think we could get rid of `Empty()` and replace with `Equals(...)`

I'll make a follow up ticket for this

Reply:
> So I think we could get rid of Empty() and replace with Equals(...)

Depending on how we leverage `comparable` constriants, both of those could be eliminated:
```go
func isZero[C comparable](val C) bool {
	var zero C
	return zero == val
}
func equal[C comparable](val, other C) bool {
	return val == other
}
```

Reply:
I would like to remove it in a follow-up PR, and move that code that needs it, outside of the Txmgr.
Is that ok?